src/utils/lagrange_poly.rs

```diff
@@ -0,0 +1,47 @@
+use ark_ff::PrimeField;
+use ark_poly::{univariate::DensePolynomial, DenseUVPolynomial};
+
+/// Computes the lagrange interpolated polynomial from the given points `p_i`
+pub fn compute_lagrange_interpolated_poly<F: PrimeField>(p_i: &[F]) -> DensePolynomial<F> {
+    // TODO: build domain directly from field, avoid explicit conversions within the loop
+
+    // domain is 0..p_i.len(), to fit `interpolate_uni_poly` from hyperplonk
+    let domain: Vec<usize> = (0..p_i.len()).collect();
+
+    // compute l(x), common to every basis polynomial
+    let mut l_x = DensePolynomial::from_coefficients_vec(vec![F::ONE]);
+    for x_m in domain.clone() {
+        let prod_m = DensePolynomial::from_coefficients_vec(vec![-F::from(x_m as u64), F::ONE]);
+        l_x = &l_x * &prod_m;
+    }
+
+    // compute each w_j - barycentric weights
+    let mut w_j_vector: Vec<F> = vec![];
+    for x_j in domain.clone() {
+        let mut w_j = F::ONE;
+        for x_m in domain.clone() {
+            if x_m != x_j {
+                let prod = (F::from(x_j as u64) - F::from(x_m as u64))
+                    .inverse()
+                    .unwrap();
```

Comment:
Could we remove this `.unwrap()`? to avoid panicking inside the methods. Maybe you can return `Result<DensePolynomial<F>, Error>` and then do `[...].inverse()?;`

Reply:
After checking the `.inverse()` method from arkworks, I'm thinking:
it should never be reached, since `x_m` will not be equal to `x_j`, thus `x_j-x_m` will never lead to `0`, which is what would trigger the error (unless one of them overflows the field, which should not be possible because they come from u64, and our field is much bigger).
So maybe another option, instead of returning error in this case is to keep the `.unwrap()` but adding a comment that explains this reasoning, as the `unwrap` should never be reached because of our logic, si if it is reached it should 'trigger all the alarms'.

So we have tow options:
a) removing the unwrap and returning an error if this fails
b) keeping the unwrap but adding a clarifying comment explaining that it should never fail when unwrapping (and hoping that arkworks does not change the underlying logic)

I think that is better to not depend on that arkworks does not change the underlying logic and someday unexpectedly we get a panic because the unwrap panics due a new underlying reason. But it should not be a very plausible case, so either option (a, b) is fine for me ^^

Reply:
Ok went for commenting [here](https://github.com/privacy-scaling-explorations/folding-schemes/commit/bec49e5d48d375d4ea72a878def1d0aae43200f5). Also, I think you were right on not adding error handling for the case where a user wants to lagrange interpolate more than `2^64 - 1` points. This def complexifies things API wise for not much value add.